packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -275,6 +416,18 @@ export class QueryBuilderPropertyExpressionState {
       ) {
         requiresExistsHandling = true;
       }
+      if (
+        currentExpression.func.genericType.value.rawType instanceof Class &&
+        currentExpression.func.owner._generatedMilestonedProperties.length !== 0
```

Comment:
Is it possible for the `_generatedMilestonedProperties` to be populated but the currentExpression not be a milestone expression? we should make sure its there in the list before replacing the current expression.func.